### PR TITLE
fix: redact password in parseConfigError cause and quoted DSN passwords

### PR DESCRIPTION
**Pull request by vahid-sohrabloo**
## Problem

`parseConfigError.Error()` redacts the connection string it echoes, then appends the wrapped cause verbatim:

```go
return fmt.Sprintf("cannot parse `%s`: %s (%s)", connString, e.msg, e.err.Error())
```

For a URL-form DSN the cause is a `*url.Error` that carries the **raw** connection string, so the password reappears in full immediately after being redacted:

```
cannot parse `clickhouse://ch01:9000 password=xxxxx`: failed to parse as URL
(parse "clickhouse://ch01:9000 password=PLAINSECRET": invalid port ":9000 password=PLAINSECRET" after host)
```

Redaction visibly worked on the left; the secret is right there on the right. A malformed `clickhouse://` DSN is enough to trigger it — no unusual password required — and since `chpool.ParseConfig` and `chpool.New` return this error unwrapped, every caller that logs it inherits the disclosure.

There is a second, independent defect: `password='[^']*'` does not account for the backslash escapes `parseDSNSettings` accepts inside quotes. A quoted password containing `\'` followed by a space defeats that pass, and the following `password=[^ ]*` pass stops at the space, leaving the tail exposed. This one leaks through the redactor itself, so it affects keyword DSNs too.

| DSN | leaked before |
| --- | --- |
| `clickhouse://ch01:9000 password=PLAINSECRET` | yes — via cause |
| `clickhouse://alice:PLAINSECRET@host:notaport` | yes — via cause |
| `host=ch port=notanumber user=alice password='se\'x PLAINSECRET'` | yes — via redactor |
| `host=ch port=notanumber user=alice password=PLAINSECRET` | no |

## Fix

- Route the wrapped cause through the same redaction before interpolating it into the message.
- Extend the quoted-password pattern to `password='(?:[^'\\]|\\.)*'` so escaped quotes no longer terminate the match early.
- Hoist the three patterns to package-level vars now that two call sites share them, which also stops them being recompiled on every error.

`Unwrap()` still returns the original cause, so `errors.Is`/`errors.As` behaviour is unchanged. Callers that deliberately reach through `errors.Unwrap(err).Error()` can still see the raw text; wrapping with `%w` and formatting goes through the redacted `Error()` and is safe.

## Tests

`TestConfigError` previously passed `nil` as the cause in every case, which is why this path was never covered. Added cases for the cause-bearing path and for the escaped-quote password, plus `TestParseConfigErrorDoesNotLeakPassword`, which drives the four DSNs above through the real `ParseConfig` and asserts the secret is absent from `Error()`.

All four cases fail before this change and pass after. Full suite (`go test ./...`) passes against a local ClickHouse, and `golangci-lint run ./` reports 0 issues.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved configuration error messages to consistently redact passwords and other sensitive connection details.
  * Prevented secrets from leaking through wrapped or malformed connection errors, including cases where underlying errors may echo connection strings.
  * Enhanced redaction behavior for passwords containing escaped quotes and other punctuation edge cases.
  * Added test coverage to verify no raw secret is present in returned error text.

### CODE REVIEW

File: errors.go [hunk 1]
```diff
@@ -245,19 +247,31 @@ func (e *writeError) Unwrap() error {
...
+	return redactSecrets(connString)
+}
+
+// redactSecrets strips password values out of arbitrary text. It is used both
+// for connection strings that could not be parsed and for the wrapped errors
+// that quote them back.
+func redactSecrets(s string) string {
+	s = quotedDSNPW.ReplaceAllLiteralString(s, "password=xxxxx")
+	s = plainDSNPW.ReplaceAllLiteralString(s, "password=xxxxx")
+	s = brokenURLPW.ReplaceAllLiteralString(s, ":xxxxxx@")
```
[the remark this answers is not in the document]
Comment (vahid-sohrabloo) [reply errors.go]: Confirmed and fixed in b79f996 — but with a broader pattern than proposed, because the suggested one doesn't close the whole hole.

`[^@]+` cannot span an `@`, so it fixes the colon case and leaves the at-sign case exactly as it was. Measured against the three candidates:

| input | current | proposed `[^@]+@` | shipped `\S*@` |
| --- | --- | --- | --- |
| `alice:secret:tail@host` | leaks `secret` | fixed | fixed |
| `alice:pa@ss@host` | leaks `ss` | **still leaks `ss`** | fixed |
| `user:pass/word@host` | redacted | redacted | redacted |

Both cases are reachable for the same reason: the delimiter in the password is *what makes `url.Parse` fail*, which is precisely what routes the string to this fallback instead of `redactURL`. Go splits the authority at the last `@` and then rejects the userinfo, so `alice:pa@ss@host` never reaches `redactURL`.

Shipped `(://[^/@\s:]+:)\S*@` with `${1}xxxxxx@`. Two notes on the character class:

- `\S` rather than `[^\s/]`. Excluding `/` looks tighter but regresses the existing `weird url with slash in password` case (`clickhouse://user:pass/word@host:5432/db_name`) to fully unredacted — the run stops before `/db_name` and never reaches an `@`. Verified that variant against the suite before discarding it.
- Anchoring on `://` has a side benefit: the old pattern mangled `clickhouse://other@host/db` into `clickhouse:xxxxxx@host/db` when it reached this path. That's now left intact.

Regression cases added to `TestParseConfigErrorDoesNotLeakPassword` for both delimiters; both fail before the change and pass after. Full suite green across all 12 packages, `golangci-lint run ./` reports 0 issues.

Residual, not addressed: greedy-to-last-`@` can over-redact when the surrounding error text puts another `@` in the same whitespace-free token, swallowing host and path too. That direction is safe — it errs toward hiding more, never less.
